modules/lib/core/index.d.ts

```diff
@@ -142,6 +142,318 @@ export interface ScriptValue {
+    ? A
+    : Pick<A, Exclude<keyof A, keyof B>> & B;
+
+export interface ComplexCookie {
+    /**
+     * The value of the cookie (optional).
+     * 
+     * @type string
+     */
+    value: string
+    /**
+     * A comment (rfc2109) to document the cookie (optional).
+     * 
+     * @type string
+     */
+    comment?: string
+    /**
+     * The expiration date and time for the cookie (optional).
+     *
+     * @type string
+     */
+    expires?: Date;
+    /**
+     * The domain name for which the cookie is set.
+     *
+     * @type string
+     */
+    domain?: string
+    /**
+     *  Indicates whether the cookie should not be accessible via JavaScript (optional).
+     *
+     * @type string
+     */
+    httpOnly?: boolean
+    /**
+     * The maximum age of the cookie in seconds (optional).
+     *
+     * @type string
+     */
+    maxAge?: number
+    /**
+     * The path on the server where the cookie should be available.
+     *
+     * @type string
+     */
+    path?: string
+    /**
+     *  Specifies the SameSite attribute (draft RFC) for the cookie (optional).
+     *
+     * @type string
+     */
+    sameSite?: LiteralUnion<'lax' | 'strict' | 'none'>;
+    /**
+     * Indicates whether the cookie should only be sent over HTTPS (optional).
+     *
+     * @type string
+     */
+    secure?: boolean
+}
+
+export type RequestBranch = 'draft' | 'master';
+export type RequestGetHeaderFunction = (headerName: string) => string | null;
+export type RequestMethod = 'GET' | 'POST' | ' HEAD' | 'OPTIONS' | ' PUT' | 'DELETE' | ' TRACE' | 'CONNECT' | ' PATCH' | 'PROPFIND' | ' PROPPATCH' | 'MKCOL' | ' COPY' | 'MOVE' | ' LOCK' | 'UNLOCK';
+export type RequestMode = 'edit' | 'inline' | 'live' | 'preview' | 'admin';
+export type RequestParams = Record<string, string | string[]>;
+export type RequestScheme = 'http' | 'https';
+
+export type RequestCookies = Record<string, string | undefined>;
+export type ResponseCookies = Record<string, string | ComplexCookie | undefined>;
+
+export type RequestHeaders = Record<string, string | undefined>;
+export type ResponseHeaders = Record<string, string | number | (string | number)[] | undefined>;
+
+export interface DefaultRequestCookies extends RequestCookies {
+    enonic_xp_tour?: string
+    JSESSIONID?: string
+}
+
+export interface DefaultRequestHeaders extends RequestHeaders {
+    Accept?: string
+    'Accept-Charset'?: string
+    'Accept-Encoding'?: string
+    'Accept-Language'?: string
+    Authorization?: string
+    'Cache-Control'?: string
+    Connection?: string
+    'Content-Length'?: string
+    'Content-Type'?: string
+    Cookie?: string
+    Language?: string
+    Host?: string
+    'If-None-Match'?: string
+    Referer?: string
+    'sec-ch-ua'?: string
+    'sec-ch-ua-mobile'?: string
+    'sec-ch-ua-platform'?: string
+    'Sec-Fetch-Dest'?: string
+    'Sec-Fetch-Mode'?: string
+    'Sec-Fetch-Site'?: string
+    'Sec-Fetch-User'?: string
```

Comment:
some experimental headers whe really never use. Why are they int default? Why they are different case BTW?

Reply:
Default optional, only for autocomplete. These are the ones in chrome.

Dunno what to use them for on the server-side, so probably can be removed.

Are they in the way though?

Reply:
I removed the experimental ones and added autocompletion to the Sec-Fetch ones. 

I also added Priority since both Chrome and Firefox sends it.

There are tons of headers:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers
https://en.wikipedia.org/wiki/List_of_HTTP_header_fields

Not really sure which ones should be part of DefaultRequestHeaders.